src/components/admin/article/ArticleController.jsx

```diff
@@ -75,19 +87,76 @@ class ArticleController extends FalcorController {
       );
     }
 
-    Promise.all(updatePromises).then(() => {
-      // Reset state after save is done
-      this.safeSetState({
-        changed: false,
-        refresh: true,
-        authorsAdded: [],
-        authorsDeleted: {},
-        changesObject: { mainForm: false, authors: false },
+    new Promise(resolve => {
```

Comment:
1. it would be nice to write this with async/await, think it'd become a lot cleaner. 

2. Do you think it would be possible to move a lot of the responsibility here to the backend? Could we just put all this logic in the `updateTags` falcor call? So that it checks if the id is null etc. and if it is create it etc.

Reply:
Yup, 1), and 2) fixed.